bigtable-hbase-replication/src/main/java/com/google/cloud/bigtable/hbase/replication/HbaseToCloudBigtableReplicationEndpoint.java

```diff
@@ -165,25 +165,23 @@ public boolean replicate(ReplicateContext replicateContext) {
           entries.stream()
               .collect(groupingBy(entry -> entry.getKey().getTablename().getNameAsString()));
 
-      for (Map.Entry<String, List<WAL.Entry>> walEntriesForTable : walEntriesByTable.entrySet()) {
-        final String tableName = walEntriesForTable.getKey();
-        // TODO: MAKE IT PARALLEL
+      walEntriesByTable.entrySet().parallelStream().forEach(walEntries -> {
```

Comment:
We should control the number of threads used in this endpoint explicitly. There can be any number of replciated tables and there can be more than 1 endpoint object, 

We should create a fixed size threadpool and force all the tables to replicate on that threadpool. That way we can limit the concurrency on the region server.